pennylane/tensorbox/fn.py

```diff
@@ -0,0 +1,416 @@
+
+    **Example**
+
+    >>> x = torch.tensor([1., 2.])
+    >>> get_interface(x)
+    'torch'
+    >>> from pennylane import numpy as np
+    >>> x = np.array([4, 5], requires_grad=True)
+    >>> get_interface(x)
+    'autograd'
+    """
+    return TensorBox(tensor).interface
+
+
+def toarray(tensor):
+    """Returns the tensor as a NumPy ``ndarray``. No copying
+    is performed; the tensor and the returned array share the
+    same storage.
+
+    Args:
+        tensor (tensor_like): input tensor
+
+    Returns:
+        array: a ``ndarray`` view into the same data
+
+    **Example**
+
+    >>> x = torch.tensor([1., 2.])
+    >>> toarray(x)
+    array([1, 2])
+    """
+    return TensorBox(tensor).numpy()
+
+
+def ones_like(tensor, dtype=None):
+    """Returns a tensor of all ones with the same shape and dtype
+    as the input tensor.
+
+    Args:
+        tensor (tensor_like): input tensor
+        dtype (str, np.dtype): The desired output datatype of the array. If not provided the dtype of
+            ``tensor`` is used. This argument can be any supported NumPy dtype representation, including
+            a string (``"float64"``), a ``np.dtype`` object (``np.dtype("float64")``), or
+            a dtype class (``np.float64``). If ``tensor`` is not a NumPy array, the
+            **equivalent** dtype in the dispatched framework is used.
+
+    Returns:
+        tensor_like: an all-ones tensor with the same shape and
+        size as ``tensor``
+
+    **Example**
+
+    >>> x = torch.tensor([1., 2.])
+    >>> ones_like(x)
+    tensor([1, 1])
+    >>> y = tf.Variable([[0], [5]])
+    >>> ones_like(y, dtype=np.complex128)
+    <tf.Tensor: shape=(2, 1), dtype=complex128, numpy=
+    array([[1.+0.j],
+           [1.+0.j]])>
+    """
+    if dtype is not None:
+        return TensorBox(tensor).ones_like().cast(dtype).data
+
+    return TensorBox(tensor).ones_like().data
+
+
+def requires_grad(tensor):
+    """Returns True if the tensor is considered trainable.
+
+    .. warning::
+
+        The implemetation depends on the contained tensor type, and
+        may be context dependent.
+
+        For example, Torch tensors and PennyLane tensors track trainability
+        as a property of the tensor itself. TensorFlow, on the otherhand,
+        only tracks trainability if being watched by a gradient tape.
+
+    Args:
+        tensor (tensor_like): input tensor
+
+    **Example**
+
+    Calling this function on a PennyLane NumPy array:
+
+    >>> x = np.array([1., 5.], requires_grad=True)
+    >>> qml.tensorbox.requires_grad(x)
+    True
+    >>> x.requires_grad = False
+    >>> qml.tensorbox.requires_grad(x)
+    False
+
+    PyTorch has similar behaviour.
+
+    With TensorFlow, the output is dependent on whether the tensor
+    is currently being watched by a gradient tape:
+
+    >>> x = tf.Variable([0.6, 0.1])
+    >>> requires_grad(x)
+    False
+    >>> with tf.GradientTape() as tape:
+    ...     print(requires_grad(x))
+    True

```

Comment:
just a thought...wouldn't it be important to support the same call structure in all frameworks? if i don't know which tensor type enters a template, how do i know whether to call `requires_grad()` or `.requires_grad`?